pkg/landscaper/controllers/deployitem/actuator.go

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package deployitem
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/apis/core/v1alpha1/helper"
+	"github.com/gardener/landscaper/pkg/landscaper/deployitem"
+)
+
+// NewActuator creates a new deploy item controller that handles timeouts
+// To detect pickup timeouts (when a DeployItem resource is not reconciled by any deployer within a specified timeframe), the controller checks for a timestamp annotation.
+// It is expected that deployers remove the timestamp annotation from deploy items during reconciliation. If the timestamp annotation exists and is older than a specified duration,
+// the controller marks the deploy item as failed.
+// rawPickupTimeout is a string containing the pickup timeout duration, either as 'none' or as a duration that can be parsed by time.ParseDuration.
+func NewActuator(log logr.Logger, c client.Client, scheme *runtime.Scheme, rawPickupTimeout string) (reconcile.Reconciler, error) {
+	act := actuator{log: log, c: c, scheme: scheme}
+	if rawPickupTimeout == "none" {
+		act.pickupTimeout = nil
+	} else if len(rawPickupTimeout) == 0 {
+		tmp := deployitem.DefaultPickupTimeout
+		act.pickupTimeout = &tmp
+	} else {
+		tmp, err := time.ParseDuration(rawPickupTimeout)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse deploy item pickup timeout into a duration: %w", err)
+		}
+		act.pickupTimeout = &tmp
+	}
+
+	// log pickup timeout
+	timeoutLog := ""
+	if act.pickupTimeout != nil {
+		timeoutLog = act.pickupTimeout.String()
+	}
+	log.Info("deploy item pickup timeout detection", "active", act.pickupTimeout != nil, "timeout", timeoutLog)
+
+	return &act, nil
+}
+
+type actuator struct {
+	log           logr.Logger
+	c             client.Client
+	scheme        *runtime.Scheme
+	pickupTimeout *time.Duration
+}
+
+func (a *actuator) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := a.log.WithValues("resource", req.NamespacedName.String())
```

Comment:
```suggestion
        if a.pickupTimeout == nil {
          return nil
        }
	logger := a.log.WithValues("resource", req.NamespacedName.String())
```
a not set pickup timeout should be handled

Reply:
It is handled, but in a different way:
It is possible to configure the landscaper to not do any pickup timeout checks by setting the timeout in the config to `none`. In this case, `a.pickupTimeout` will be set to `nil`, otherwise it will contain a `time.Duration`.